File: vortexdbg-api/src/main/java/com/vortexdbg/thread/ThreadDispatcher.java
package com.vortexdbg.thread;

import com.vortexdbg.signal.SignalOps;
import com.vortexdbg.signal.SignalTask;

import java.util.List;
import java.util.concurrent.TimeUnit;

public interface ThreadDispatcher extends SignalOps {

    void addThread(ThreadTask task);

    List<Task> getTaskList();

    Number runMainForResult(MainTask main);

    void runThreads(long timeout, TimeUnit unit);

    int getTaskCount();

    boolean sendSignal(int tid, int sig, SignalTask signalTask);

    RunnableTask getRunningTask();

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/thread/ThreadDispatcher.kt
package com.vortexdbg.thread

import com.vortexdbg.signal.SignalOps
import com.vortexdbg.signal.SignalTask
import java.util.concurrent.TimeUnit

interface ThreadDispatcher : SignalOps {

    fun addThread(task: ThreadTask)

    fun getTaskList(): List<Task>

    fun runMainForResult(main: MainTask): Number?

    fun runThreads(timeout: Long, unit: TimeUnit)

    fun getTaskCount(): Int

    fun sendSignal(tid: Int, sig: Int, signalTask: SignalTask?): Boolean

    fun getRunningTask(): RunnableTask?

}
